internal/subscriber/model.go

```diff
@@ -151,25 +170,68 @@ func (a *AckMessage) HasHitDeadline() bool {
 }
 
 // ParseAckID ...
-func ParseAckID(ackID string) *AckMessage {
+func ParseAckID(ctx context.Context, ackID string) (*AckMessage, error) {
 	// split the message and parse tokens
 	parts := strings.Split(ackID, ackIDSeparator)
+	decodedParts := decode(parts)
+
+	if !isValidAckID(decodedParts) {
+		err := merror.Newf(merror.InvalidArgument, "AckID received is not in expected format")
+		logger.Ctx(ctx).Errorw(err.Error())
+		return nil, err
+	}
+
+	partition, err := strconv.ParseInt(decodedParts[3], 10, 0)
+	if err != nil {
+		logger.Ctx(ctx).Errorw(err.Error())
+		err = fmt.Errorf("error in parsing partition value %v to int", decodedParts[3])
```

Comment:
try using the below instead and see if this helps create a more descriptive error.
```
errors.WithMessage(err , fmt.Sprintf("error in parsing partition value %v to int", decodedParts[3]))
```

And no need to log here if the calling function is doing this. Can remove the duplicate log lines. In turn, can remove ctx as well. Do check the file for more such instances.

Reply:
Done